### PR TITLE
The supported models detection improvement

### PR DESCRIPTION
**Pull request by vladislav-volkov**
During model loading, the XML file is parsed twice.
First time, while the model version is checked. Second time while model is loading.
This fix allows to avoid model file twice parsing.

### CODE REVIEW

File: inference-engine/src/readers/ir_reader/ie_ir_reader.cpp [hunk 1]
```diff
@@ -27,29 +28,29 @@ bool IRReader::supportModel(std::istream& model) const {
...
     std::string header(header_size, ' ');
     model.read(&header[0], header_size);
+    model.seekg(0, model.beg);
+
+    // Find '<net ' substring and version in the .xml file
+    // Regular expression format:
+    //  (Opional comments or spaces)
+    //      and (<net name="string" version="number"> or <net version="number" name="string">)
+    //      and (Opional comments or spaces)
+    std::regex regex(R"rs((?:(?:<!--[\s\S]*-->)|\s)*(?:<[nN]et\s*name\s*=\s*"(?:[^<&"])*"\s*version\s*=\s*"\s*(\d*)\s*"\s*>|<[nN]et\s*version\s*=\s*"\s*(\d*)\s*"\s*name\s*=\s*"(?:[^<&"])*"\s*>)(?:(?:<!--[\s\S]*-->)|\s)*)rs",
```
Comment (ilya-lavrenov) [inline inference-engine/src/readers/ir_reader/ie_ir_reader.cpp:38]: regex is not supported on CentOS with obsolete compiler. Can we extract the version part of IR until the `>` (it's a `<net ... version="10">`) and parse via pugixml?

**Comment by ilyachur** [comment]
Some tests are failing:

> [ RUN      ] CNNNetReaderImplTest.ReadInThreads
> terminate called after throwing an instance of 'InferenceEngine::details::InferenceEngineException'
>   what():  Unknown model format! Cannot find reader for the model and read it. Please check that reader library exists in your PATH.
> terminate called recursively

File: inference-engine/src/readers/ir_reader/ie_ir_reader.cpp [hunk 2]
```diff
@@ -27,20 +29,21 @@ bool IRReader::supportModel(std::istream& model) const {
     const int header_size = 128;
     std::string header(header_size, ' ');
     model.read(&header[0], header_size);
+    model.seekg(0, model.beg);
+
+    pugi::xml_document doc;
+    auto res = doc.load_string(header.c_str(), pugi::parse_default | pugi::parse_fragment);
```
Comment (ilyachur) [inline inference-engine/src/readers/ir_reader/ie_ir_reader.cpp:35]: Am I right that header contains some incorrect string? Something like:
`<net version="10" name="test"> <layers><lay`
or
`<net version="10" name="test"`
or
`<net name="test" versio`

Will it works for all cases?

Comment (ilya-lavrenov) [reply inference-engine/src/readers/ir_reader/ie_ir_reader.cpp:35]: BTW, may be we need to extract this function (which detects version of IR) to a separate header and write unit tests for it.

Comment (vladislav-volkov) [reply inference-engine/src/readers/ir_reader/ie_ir_reader.cpp:35]: Yes, all of these cases are handled correctly.

File: inference-engine/src/readers/ir_reader/ie_ir_reader.cpp (hunk 1 above)
Comment (vladislav-volkov) [reply inference-engine/src/readers/ir_reader/ie_ir_reader.cpp]: Fixed

File: inference-engine/tests/functional/inference_engine/net_reader_test.cpp [hunk 3]
```diff
@@ -146,6 +146,71 @@ TEST_P(NetReaderTest, ReadCorrectModelWithWeightsUnicodePath) {
...
+        <edge from-layer="0" from-port="0" to-layer="1" to-port="1"/>
+        <edge from-layer="1" from-port="2" to-layer="2" to-port="0"/>
+    </edges>
+</net>
+)V0G0N";
+
+    std::string headers[] = {
+        R"()",
+        R"(<!-- <net name="Network" version="100500"> -->)",
+        R"(<!-- <net name="Network" version="10" some_attribute="Test Attribute"> -->)"
```
Comment (ilyachur) [inline inference-engine/tests/functional/inference_engine/net_reader_test.cpp:204]: Could you add comments about what covered each case?

Comment (ilya-lavrenov) [reply inference-engine/tests/functional/inference_engine/net_reader_test.cpp:204]: can be done later not to trigger CI one more time :)

**Pull request merged**